src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,202 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+'''
+Provide a PopWindow class for pop-up windows.
+Making PopWindow containing Gtk.Toolbar which also contains Gtk.Label
+and Toolbutton at the end of the Gtk.Toolbar.
+
+It is possible to change props like size and add more widgets PopWindow
+and also to Gtk.Toolbar.
+
+Example:
+    .. literalinclude: ..sugar/src/jarabe/view/viewsource.py
+    .. literalinclude: ..sugar/src/jarabe/view/viewhelp.py
+'''
+from gettext import gettext as _
+from gi.repository import Gtk
+from gi.repository import Gdk
+from gi.repository import GdkX11
+from gi.repository import GObject
+
+from sugar3.graphics import style
+from sugar3.graphics.toolbutton import ToolButton
+
+from jarabe.model import shell
+
+
+class PopWindow(Gtk.Window):
+    """
+    UI interface for activity Pop-up Windows.
+    PopWindows are the windows that open on the top of the current window.
+    These pop-up windows don't cover the whole screen.
+    They contain canvas content, alerts messages, a tray and a
+    toolbar.
+
+    FULLSCREEN and HALF_WIDTH for setting size of the window.
+
+    Kwargs:
+        size (int,int): size to be set of the window
+        window_xid (xlib.Window): xid of the parent window
+    """
+    FULLSCREEN = (Gdk.Screen.width() - style.GRID_CELL_SIZE * 3,
+                  Gdk.Screen.height() - style.GRID_CELL_SIZE * 2)
+
+    HALF_WIDTH = ((Gdk.Screen.height() - style.GRID_CELL_SIZE * 3)/2,
+                  (Gdk.Screen.height() - style.GRID_CELL_SIZE * 2))
+
+    def __init__(self, window_xid=None, **kwargs):
+        Gtk.Window.__init__(self, **kwargs)
+        self._parent_window_xid = window_xid
+
+        self.set_decorated(False)
+        self.set_position(Gtk.WindowPosition.CENTER_ALWAYS)
+        self.set_border_width(style.LINE_WIDTH)
+        self.set_has_resize_grip(False)
+        self.props.size = self.FULLSCREEN
+
+        self.connect('realize', self.__realize_cb)
+        self.connect('key-press-event', self.__key_press_event_cb)
+        self.connect('hide', self.__hide_cb)
+
+        self._vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
+        self.add(self._vbox)
+        self._vbox.show()
+
+        self._title_box = TitleBox()
+        self._title_box.close_button.connect(
+            'clicked',
+            self.__close_button_clicked_cb)
+        self._title_box.set_size_request(-1, style.GRID_CELL_SIZE)
+
+        self._vbox.pack_start(self._title_box, False, True, 0)
+        self._title_box.show()
+
+        # Note:
+        # Not displaying the pop-up from here instead allowing
+        # the child class to display the window after modifications
+        # like chaninging window size, decorating, changing position.
+
+    def set_size(self, size):
+        width, height = size
```

Comment:
This way of assigning two variable with one value isn't right. It should be somewhat like this ```width, height = size_x, size_y```